R/equivalence_test.R

```diff
@@ -36,30 +38,51 @@ equivalence_test <- function(posterior, bounds = "default", CI = 90, verbose = T
 
 #' @export
 equivalence_test.numeric <- function(posterior, bounds = "default", CI = 90, verbose = TRUE) {
-  rope_value <- rope(posterior, bounds = bounds, CI = CI)
-  if ("rope" %in% class(rope_value)) {
-    rope_value <- as.numeric(rope_value)
-  } else {
-    rope_value <- sapply(rope_value, as.numeric)
-  }
-  decision <- ifelse(rope_value == 0, "rejected",
-    ifelse(rope_value == 100, "accepted", "undecided")
+  out <- as.data.frame(rope(posterior, bounds = bounds, CI = CI))
+
+  out$ROPE_Equivalence <- ifelse(out$ROPE_Percentage == 0, "rejected",
+    ifelse(out$ROPE_Percentage == 100, "accepted", "undecided")
   )
+  return(out)
+}
+
 
-  if (length(CI) > 1) {
-    decision <- split(unname(decision), names(decision))
-  }
 
-  decision
+
+#' @importFrom stats sd
+#' @keywords internal
+.equivalence_test_models <- function(posterior, bounds = "default", CI = 90, verbose = TRUE) {
+  if (all(bounds == "default")) {
+    bounds <- c(-0.1 * sd(insight::get_response(posterior)), 0.1 * sd(insight::get_response(posterior)))
+  } else if (!all(is.numeric(bounds)) | length(bounds) != 2) {
+    stop("bounds should be 'default' or a vector of 2 numeric values (e.g., c(-0.1, 0.1)).")
+  }
+  list <- sapply(get_parameters(posterior), equivalence_test, bounds = bounds, CI = CI, verbose = verbose, simplify = FALSE)
+  return(flatten_list(list, name = "Parameter"))
 }
 
+#' @export
+equivalence_test.stanreg <- .equivalence_test_models
+
+#' @export
+equivalence_test.brmsfit <- .equivalence_test_models
+
+
+
+
 
+#' TEMPORARY
+#' @importFrom stats coef
+#' @param model model
 #' @export
-equivalence_test.stanreg <- function(posterior, bounds = "default", CI = 90, verbose = TRUE) {
-  sapply(as.data.frame(posterior), equivalence_test, bounds = bounds, CI = CI, verbose = verbose, simplify = FALSE)
+find_parameters <- function(model) {
+  return(names(coef(model)))
 }
 
+#' TEMPORARY
+#' @importFrom stats coef
+#' @param model model
 #' @export
-equivalence_test.brmsfit <- function(posterior, bounds = "default", CI = 90, verbose = TRUE) {
-  sapply(as.data.frame(posterior), equivalence_test, bounds = bounds, CI = CI, verbose = verbose, simplify = FALSE)
+get_parameters <- function(model) {
```

Comment:
get_parameters and find_parameters were temporarily put here (related to [this](https://github.com/easystats/insight/issues/24))

Reply:
Have you updated the gitignore file on your computer? I'll take a look at this later, it's indeed a bit too messy to inspect from mobile phone. ;-)